### ISSUE TITLE
[Feature] Use component init/create instead of component run?

### ISSUE THREAD

**Issue by resouer** [oam-dev/kubevela#330]
It seems component init/create are all better than component run?

Normally, we only say workload run (which was the old idea of vela but we now change it to component).

Also, we may need to consider have component deploy besides component init/create.

**Comment by wonderflow** [comment]
If so, I would take all component level command as staging command, these changes will take affect only after `vela app deploy` executed.

**Comment by resouer** [comment]
> If so, I would take all component level command as staging command, these changes will take affect only after `vela app deploy` executed.

@wonderflow either `vela app deploy` or even `vela deploy` works for me. Just don't do component run.

**Comment by wonderflow** [comment]
I want to clarify more if we make this change.

Adding a `vela deploy` command means we will make every change take real effect only by this command.

This contains two aspects:

* `vela comp init`(Workload) will only affect local appfile, the serverside AppConfig won't change.
* `vela route`(Trait) will only affect local appfile, the  serverside AppConfig won't change.

Only `vela deploy` will make serverside AppConfig change.

Do you mean that? @resouer

**Comment by ryanzhang-oss** [comment]
Just to summarize the discussion here. I think there are two flavors. 

1.  One-step operation, aka. `run`. The changes apply to the server side as soon as the command is issued
2. Two-step operation.  aka. `init`/`create` followed by a `deploy` command

IMO, I think we might want to support both and if we want to support one first, I would vote for the two-step appraoch.

**Comment by resouer** [comment]
@ryanzhang-oss I am fine to support one-step flavor only temporarily. I'm just proposing to use sth else instead of `run` for component deploy. i.e.:

I am proposing:

```console
$ vela comp deploy -i nginx -p 80 --name frontend # init and deploy
``` 

What we will add in the future (low priority):

```console
$ vela comp init -i nginx -p 80 --name frontend # init only

$ vela scale frontend 100
FAIL: component frontend should be deployed before scale! Please do: `vela comp deploy frontend`.

$ vela comp deploy frontend # deploy only

$ vela scale frontend 100
SUCCESS!
```

/cc @wonderflow

**Comment by wonderflow** [comment]
@resouer Agree, in this case, we must use `vela comp deploy`, using `vela deploy` will cause confusion
